include/cetl/rtti.hpp

```diff
@@ -81,6 +82,15 @@ constexpr type_id type_id_type_value() noexcept
 template <typename T>
 constexpr type_id type_id_value = T::_get_type_id_();
 
+/// The type ID value specializations for `void` type.
+///
+/// In use for the `type_id_invalid` constant, which f.e. is in use at
+/// `unbounded_variant::type_id()` method to indicate that the variant is valueless.
+///
+template <>
+constexpr type_id type_id_value<void>{};
+constexpr type_id type_id_invalid{type_id_value<void>};
+
```

Comment:
This is not good because it breaks the orthogonality of the types. All type-ID values should be valid; if you need to represent the absence of a type-ID, use `optional`. Seeing as we can't eat our own cookies, we can define a special case for `unbounded_variant::type_id` such that it returns the type-ID of `void` if it's empty.

Essentially I'm saying that we should keep your current design but remove this special `type_id_invalid` and confine the poor typing to one method because there's no other way.